mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1374,6 +1375,63 @@ pub fn create_taker_spends_maker_payment_preimage<T: UtxoCommonOps + SwapOps>(
     Box::new(fut.boxed().compat())
 }
 
+pub fn create_taker_refunds_payment<T: UtxoCommonOps + SwapOps>(
+    coin: T,
+    taker_payment_tx: &[u8],
+    time_lock: u32,
+    maker_pub: &[u8],
+    secret_hash: &[u8],
+    swap_unique_data: &[u8],
+) -> TransactionFut {
+    let my_address = try_tx_fus!(coin.as_ref().derivation_method.iguana_or_err()).clone();
+    let mut prev_transaction: UtxoTx =
+        try_tx_fus!(deserialize(taker_payment_tx).map_err(|e| TransactionErr::Plain(format!("{:?}", e))));
+    prev_transaction.tx_hash_algo = coin.as_ref().tx_hash_algo;
+    drop_mutability!(prev_transaction);
+    if prev_transaction.outputs.is_empty() {
+        return try_tx_fus!(TX_PLAIN_ERR!("Transaction doesn't have any output"));
+    }
+
+    let key_pair = coin.derive_htlc_key_pair(swap_unique_data);
+    let script_data = Builder::default().push_opcode(Opcode::OP_1).into_script();
+    let redeem_script = payment_script(
+        time_lock,
+        secret_hash,
+        key_pair.public(),
+        &try_tx_fus!(Public::from_slice(maker_pub)),
+    )
+    .into();
+    let fut = async move {
+        let fee = try_tx_s!(coin.get_htlc_spend_fee(DEFAULT_SWAP_TX_SPEND_SIZE).await);
```

Comment:
I don't think that it should be done, but I think we need to increase the fee as it can be increased until the watcher broadcasts this transaction.
This is the for discussion.

There is the following method
https://github.com/KomodoPlatform/atomicDEX-API/blob/fb48668f4dc6b9645661551e1be683f93550e53d/mm2src/coins/utxo.rs#L842-L844
for such purposes. It takes `FeeApproxStage`
https://github.com/KomodoPlatform/atomicDEX-API/blob/fb48668f4dc6b9645661551e1be683f93550e53d/mm2src/coins/lp_coins.rs#L1032-L1041
that can be extended with `WatcherPreimage`, for example.

Reply:
Good point, this also applies to `taker_spends_maker_payment` for watchers which also uses a preimage. I'll look into it.